internal/core/logger.go

```diff
@@ -0,0 +1,54 @@
+package core
+
+import (
+	"log"
+	"os"
+)
+
+// ConfigLogger is the key for the pipeline's logger
+const ConfigLogger = "Core.Logger"
+
+// Logger defines the output interface used by Hercules components.
+type Logger interface {
+	Info(...interface{})
+	Infof(string, ...interface{})
+	Warn(...interface{})
+	Warnf(string, ...interface{})
+	Error(...interface{})
+	Errorf(string, ...interface{})
+}
+
+// DefaultLogger is the default logger used by a pipeline, and wraps the standard
+// log library.
+type DefaultLogger struct {
+	I *log.Logger
+	W *log.Logger
+	E *log.Logger
+}
+
+// NewLogger returns a configured default logger.
+func NewLogger() *DefaultLogger {
+	return &DefaultLogger{
+		I: log.New(os.Stdout, "[INFO] ", log.LstdFlags),
```

Comment:
We should log everything to stderr by default, since stdout is used to return analysis results.